src/sage/rings/function_field/ideal.py

```diff
@@ -455,20 +455,20 @@ def divisor(self):
 
         EXAMPLES::
 
-            sage: # needs sage.rings.finite_rings
+            sage: # needs sage.modules sage.rings.finite_rings
             sage: K.<x> = FunctionField(GF(4))
             sage: O = K.maximal_order()
             sage: I = O.ideal(x*(x + 1)^2/(x^2 + x + 1))
             sage: I.divisor()
             Place (x) + 2*Place (x + 1) - Place (x + z2) - Place (x + z2 + 1)
 
-            sage: # needs sage.rings.finite_rings
+            sage: # needs sage.modules sage.rings.finite_rings
             sage: Oinf = K.maximal_order_infinite()
             sage: I = Oinf.ideal((x + 1)/(x^3 + 1))
             sage: I.divisor()
             2*Place (1/x)
 
-            sage: # needs sage.rings.function_field
+            sage: # needs sage.modules sage.rings.function_field
             sage: K.<x> = FunctionField(GF(2)); _.<T> = PolynomialRing(K)
```

Comment:
Doesn't `sage.rings.function_field` depend on `sage.modules`?

Reply:
I'm not sure right now. This dependency is not declared in sage.features, and there are quite a number of these `# needs sage.modules sage.rings.function_field` annotations already elsewhere

Reply:
`sage.rings.function_field` (the python module) depends on (uses) `sage.modules`. Then the distribution package that provides the feature `sage.rings.function_field` should also provide the feature `sage.modules`. Isn't this the design?

Reply:
Well, looking at `git grep -E 'sage.(modules|matrix).*import' src/sage/rings/function_field`, I see that most places are careful to do the imports when they are needed in individual methods.

On the other hand, the distiributions per #35095: 
```
git --no-pager grep 'sage_setup:' src/sage/rings/function_field
src/sage/rings/function_field/all.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/all__sagemath_categories.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/all__sagemath_modules.py:# sage_setup: distribution = sagemath-modules
src/sage/rings/function_field/all__sagemath_pari.py:# sage_setup: distribution = sagemath-pari
src/sage/rings/function_field/all__sagemath_singular.py:# sage_setup: distribution = sagemath-singular
src/sage/rings/function_field/all__sagemath_symbolics.py:# sage_setup: distribution = sagemath-symbolics
src/sage/rings/function_field/constructor.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/derivations.py:# sage_setup: distribution = sagemath-modules
src/sage/rings/function_field/derivations_polymod.py:# sage_setup: distribution = sagemath-singular
src/sage/rings/function_field/derivations_rational.py:# sage_setup: distribution = sagemath-modules
src/sage/rings/function_field/differential.py:# sage_setup: distribution = sagemath-modules
src/sage/rings/function_field/divisor.py:# sage_setup: distribution = sagemath-modules
src/sage/rings/function_field/element.pxd:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/element.pyx:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/element_polymod.pyx:# sage_setup: distribution = sagemath-singular
src/sage/rings/function_field/element_rational.pyx:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/extensions.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/function_field.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/function_field_polymod.py:# sage_setup: distribution = sagemath-singular
src/sage/rings/function_field/function_field_rational.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/hermite_form_polynomial.pyx:# sage_setup: distribution = sagemath-modules
src/sage/rings/function_field/ideal.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/ideal_polymod.py:# sage_setup: distribution = sagemath-singular
src/sage/rings/function_field/ideal_rational.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/maps.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/order.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/order_basis.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/order_polymod.py:# sage_setup: distribution = sagemath-singular
src/sage/rings/function_field/order_rational.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/place.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/place_polymod.py:# sage_setup: distribution = sagemath-singular
src/sage/rings/function_field/place_rational.py:# sage_setup: distribution = sagemath-categories
src/sage/rings/function_field/valuation.py:# sage_setup: distribution = sagemath-pari
```
(where **sagemath-singular** does depend on **sagemath-modules**).

So based on this, I agree with you

Reply:
I've simplified it accordingly in 7a2f540